components/registration-framework/org.wso2.carbon.identity.user.registration.engine/src/main/java/org/wso2/carbon/identity/user/registration/engine/graph/UserChoiceDecisionNode.java

```diff
@@ -0,0 +1,86 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.user.registration.engine.graph;
+
+import static org.wso2.carbon.identity.user.registration.engine.util.Constants.ErrorMessages.ERROR_CODE_UNRESOLVED_NEXT_NODE;
+import static org.wso2.carbon.identity.user.registration.engine.util.Constants.SELECTED_ACTION;
+import static org.wso2.carbon.identity.user.registration.engine.util.Constants.STATUS_COMPLETE;
+import static org.wso2.carbon.identity.user.registration.engine.util.Constants.STATUS_INCOMPLETE;
+import static org.wso2.carbon.identity.user.registration.engine.util.RegistrationFlowEngineUtils.handleClientException;
+import static org.wso2.carbon.identity.user.registration.mgt.Constants.NodeTypes.DECISION;
+import static org.wso2.carbon.identity.user.registration.mgt.Constants.StepTypes.VIEW;
+import java.util.Map;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.user.registration.engine.model.RegistrationContext;
+import org.wso2.carbon.identity.user.registration.engine.model.Response;
+import org.wso2.carbon.identity.user.registration.mgt.exception.RegistrationFrameworkException;
+import org.wso2.carbon.identity.user.registration.mgt.model.NodeConfig;
+import org.wso2.carbon.identity.user.registration.mgt.model.NodeEdge;
+
+/**
+ * Implementation of a node specific to prompting user to select a choice out of multiple registration executor options.
+ */
+public class UserChoiceDecisionNode implements Node {
+
+    private static final Log LOG = LogFactory.getLog(UserChoiceDecisionNode.class);
+
+    @Override
+    public String getName() {
+
+        return DECISION;
+    }
+
+    @Override
+    public Response execute(RegistrationContext context, NodeConfig config) throws RegistrationFrameworkException {
+
+        Map<String, String> inputData = context.getUserInputData();
+
+        if (inputData != null && inputData.containsKey(SELECTED_ACTION)) {
+            String triggeredAction = inputData.get(SELECTED_ACTION);
+            if (triggeredAction != null) {
+                for (NodeEdge edge : config.getEdges()) {
+                    if (triggeredAction.equals(edge.getTriggeringActionId())) {
+                        config.setNextNodeId(triggeredAction);
+                        break;
+                    }
+                }
+            }
+            if (config.getNextNodeId() == null) {
+                throw handleClientException(ERROR_CODE_UNRESOLVED_NEXT_NODE, config.getId(), context.getContextIdentifier());
+            }
```

Comment:
shall we refactor this to outside of the if block stating at L55. and remove the if at L69?